packages/template-retail-react-app/app/utils/marketing-cloud/marketing-cloud-email-link.js

```diff
@@ -0,0 +1,153 @@
+        })
+
+        req.write(JSON.stringify(postObject))
+
+        req.end()
+    })
+}
+
+/**
+ * Generates a unique ID for the email message.
+ *
+ * @return {string} A unique ID for the email message.
+ */
+function generateUniqueId() {
+    return crypto.randomBytes(16).toString('hex')
+}
+
+/**
+ * Sends an email to a specified contact using the Marketing Cloud API. The template email must have a
+ * `%%magic-link%%` personalization string inserted. 
+ * https://help.salesforce.com/s/articleView?id=mktg.mc_es_personalization_strings.htm&type=5
+ *
+ * @param {string} email - The email address of the contact to whom the email will be sent.
+ * @param {string} templateId - The ID of the email template to be used for the email.
+ * @param {string} magicLink - The magic link to be included in the email.
+ *
+ * @return {Promise<object>} A promise that resolves to the response object received from the Marketing Cloud API.
+ */
+async function sendMarketingCloudEmail(emailId, marketingCloudConfig) {
+    if (new Date() > marketingCloudTokenExpiration) {
+        const tokenUrl = `https://${marketingCloudConfig.subdomain}.auth.marketingcloudapis.com/v2/token`
+
+        const tokenBody = {
+            grant_type: 'client_credentials',
+            client_id: marketingCloudConfig.clientId,
+            client_secret: marketingCloudConfig.clientSecret
+        }
+
+        marketingCloudToken = (await asyncJsonHttpsPost(tokenUrl, tokenBody)).access_token
+
+        marketingCloudTokenExpiration = new Date()
+        marketingCloudTokenExpiration.setTime(
+            marketingCloudTokenExpiration.getTime() + 15 * 60 * 1000
+        )
+    }
+
+    const emailUrl = `https://${
+        marketingCloudConfig.subdomain
+    }.rest.marketingcloudapis.com/messaging/v1/email/messages/${generateUniqueId()}`
+
+    const emailHeaders = {Authorization: `Bearer ${marketingCloudToken}`}
+
+    const emailBody = {
+        definitionKey: marketingCloudConfig.templateId,
+        recipient: {
+            contactKey: emailId,
+            to: emailId,
+            attributes: {'magic-link': marketingCloudConfig.magicLink}
+        }
+    }
+
+    const emailResponse = await asyncJsonHttpsPost(emailUrl, emailBody, emailHeaders)
+
+    return emailResponse
+}
+
+/**
+ * Generates a unique ID, constructs an email message URL, and sends the email to the specified contact
+ * using the Marketing Cloud API.
+ *
+ * @param {string} email - The email address of the contact to whom the email will be sent.
+ * @param {string} templateId - The ID of the email template to be used for the email.
+ * @param {string} magicLink - The magic link to be included in the email.
+ *
+ * @return {Promise<object>} A promise that resolves to the response object received from the Marketing Cloud API.
+ */
+export async function emailLink(emailId, templateId, magicLink) {
+    const marketingCloudConfig = {
+        clientId: process.env.MARKETING_CLOUD_CLIENT_ID,
```

Comment:
I assume this will not work locally? Can we have a warning to point it out? 
```
if (!process.env.MARKKETING_CLOUD_CLIENT_SECRET) {console.warng('sth here')}
```